modules/rostests/apitests/umkm/i386/XState_asm.s

```diff
@@ -0,0 +1,163 @@
+/*
+ * PROJECT:     ReactOS API Tests
+ * LICENSE:     MIT (https://spdx.org/licenses/MIT)
+ * PURPOSE:     Assembly helpers for extended state tests
+ * COPYRIGHT:   Copyright 2025 Timo Kreuzer <timo.kreuzer@reactos.org>
+ */
+
+#include <asm.inc>
+
+#ifdef _M_IX86
+#define rcx ecx
+#define get_SSE_state @get_SSE_state@4
+#define set_SSE_state @set_SSE_state@4
+#define get_AVX_state @get_AVX_state@4
+#define set_AVX_state @set_AVX_state@4
+#define get_AVX512_state @get_AVX512_state@4
+#define set_AVX512_state @set_AVX512_state@4
+.code
+#else
+.code64
+#endif
+
+// void __fastcall get_SSE_state(__m128i data[16]);
+PUBLIC get_SSE_state
+get_SSE_state:
+    movaps [rcx + 0], xmm0
+    movaps [rcx + 16], xmm1
+    movaps [rcx + 32], xmm2
+    movaps [rcx + 48], xmm3
+    movaps [rcx + 64], xmm4
+    movaps [rcx + 80], xmm5
+    movaps [rcx + 96], xmm6
+    movaps [rcx + 112], xmm7
+#ifndef _M_IX86
+    movaps xmmword ptr [rcx + 128], xmm8
+    movaps xmmword ptr [rcx + 144], xmm9
+    movaps xmmword ptr [rcx + 160], xmm10
+    movaps xmmword ptr [rcx + 176], xmm11
+    movaps xmmword ptr [rcx + 192], xmm12
+    movaps xmmword ptr [rcx + 208], xmm13
+    movaps xmmword ptr [rcx + 224], xmm14
+    movaps xmmword ptr [rcx + 240], xmm15
+#endif
+    ret
+
+// void __fastcall set_SSE_state(__m128i data[16]);
+PUBLIC set_SSE_state
+set_SSE_state:
+    movaps xmm0, [rcx + 0]
+    movaps xmm1, [rcx + 16]
+    movaps xmm2, [rcx + 32]
+    movaps xmm3, [rcx + 48]
+    movaps xmm4, [rcx + 64]
+    movaps xmm5, [rcx + 80]
+    movaps xmm6, [rcx + 96]
+    movaps xmm7, [rcx + 112]
+#ifndef _M_IX86
+    movaps xmm8, [rcx + 128]
```

Comment:
Here and below it's not necessary to use `xmmword ptr [rcx + 128] `, contrary to the `get_SSE_state` function?

Reply:
It's not needed anywhere. Removed.